### PR TITLE
Add API helper for retrieving user problem data

### PR DESCRIPTION
**Pull request by itsjeyd**
This PR changes the analytics API client so that it can use the new API added in https://github.com/edx/edx-analytics-data-api/pull/81.

**Dependencies**

- https://github.com/edx/edx-analytics-pipeline/pull/147
- https://github.com/edx/edx-analytics-data-api/pull/81

**Partner Information**

Third-party open edX instance, not an edX partner.

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @itsjeyd! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request. 

To automatically create an OSPR issue for this pull request, just visit this link: https://openedx-webhooks.herokuapp.com/github/process_pr?number=22&repo=edx%2Fedx-analytics-data-api-client

File: analyticsclient/users.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+        Arguments:
+
+            client (analyticsclient.client.Client): The client to use to access remote resources.
+            user_id (int): The ID of the user
+
+        """
+        self.client = client
+        self.course_id = course_id
+        self.user_id = unicode(user_id)
```
Comment (bradenmacdonald) [inline analyticsclient/users.py]: Nit: I think this should only be converted to `unicode` at the time you create the URL.

**Comment by edx-webhook** [comment]
Thanks for the pull request, @itsjeyd! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request. 

To automatically create an OSPR issue for this pull request, just visit this link: https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fedx-analytics-data-api-client&number=22

File: analyticsclient/tests/test_users.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    def test_weekly_problem_data(self):
+        body = [
+            {
+                "id": 1,
+                "week_ending": "2015-08-20",
+                "course_id": self.course_id,
+                "user_id": self.user_id,
+                "problem_id": "dummy_problem_1",
+                "num_attempts": 11,
+                "final_score": '0/3',
```
Comment (bradenmacdonald) [inline analyticsclient/tests/test_users.py]: This needs to be updated.

**Comment by bradenmacdonald** [comment]
:+1:

**Comment by edx-webhook** [comment]
Thanks for the pull request, @itsjeyd! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

To automatically create an OSPR issue for this pull request, just visit this link: https://openedx-webhooks.herokuapp.com/github/process_pr?number=22&repo=edx%2Fedx-analytics-data-api-client

**Comment by edx-webhook** [comment]
Thanks for the pull request, @itsjeyd! I've created [OSPR-789](https://openedx.atlassian.net/browse/OSPR-789) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).
We can't start reviewing your pull request until you've   added yourself to the [AUTHORS](https://github.com/edx/edx-analytics-data-api-client/blob/master/AUTHORS) file . Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst) file for more information.

**Comment by edx-webhook** [comment]
Hello @itsjeyd: We are unable to continue with review of your submission at this time. Please see the associated JIRA ticket for more explanation.

**Pull request closed without merging**